SPARCbot.py

```diff
@@ -168,8 +184,276 @@ async def iwantto(self, ctx, role: str):
+        }
+        bet_log[bet_key] = new_bet
+
+        #update current_bet_id
+        bet_log['current_bet_id'] = current_bet_id
+
+        #save updated bets
+        with open("bet_log.json","w") as write_file:
+            json.dump(bet_log,write_file,indent=4)
+        return bet_log
+
+    def check_author(self,ctx,bet_id: int):
+        '''checks if the person giving commands is in the bet'''
+        #load bets
+        with open("bet_log.json", "r") as read_file:
+            bet_log = json.load(read_file)
+        bet_key = 'bet_'+str(bet_id)
+
+        #check bidder and seller (if it exists)
+        try:
+            if 'seller' in bet_log[bet_key]:
+                return bet_log[bet_key]['bidder'] == ctx.author.id or bet_log[bet_key]['seller'] == ctx.author.id
+            else:
+                return bet_log[bet_key]['bidder'] == ctx.author.id
+        except KeyError as e:
+            raise Exception('Couldn\'t check author')
+
+    @commands.command()
+    async def bet(self,ctx,*,statement):
+        '''<statement>: create an open bet'''
+        #share a unique bet_id
+        try:
+            #default bet is one-time
+            status = 'open'
+
+            #make an ongoing bet
+            if statement.startswith(config['bet_status']['standing']):
+                statement = statement.split(' ',1)[1]
+                status = 'standing'
+
+            #add the bet
+            bet_log = self.add_new_bet(ctx.author.id,statement,status)
+            bet_id = bet_log['current_bet_id']
+            nick = await get_nick_from_id(ctx,ctx.author.id)
+            await timed_send(ctx, nick+' added bet '+str(bet_id)+': \"'+statement+'\"')
+        except Exception as e:
+            await timed_send(ctx, 'Hmm...that didn\'t seem to work.')
+            raise
+
+    @commands.command()
+    async def imout(self,ctx):
+        '''Cancels your most recent unclaimed offer'''
+        if self.lock:
+            await timed_send(ctx, 'still working!')
+            return
+        self.lock = True
+
+        #load bet log
+        with open("bet_log.json", "r") as read_file:
+            bet_log = json.load(read_file)
+        current_bet_id = bet_log['current_bet_id']
+
+        #cycle through all the bets newest to oldest
+        for i in range(current_bet_id,0,-1):
+            bet_key = 'bet_'+str(i)
+            try:
+                #only find open bets the author owns
+                if bet_log[bet_key]['bidder'] == ctx.author.id and bet_log[bet_key]['status'] in ('open','standing'):
+                    removed = bet_log.pop(bet_key)
+                    with open("bet_log.json","w") as write_file:
+                        json.dump(bet_log,write_file,indent=4)
+
+                    nick = await get_nick_from_id(ctx,ctx.author.id)
+                    await timed_send(ctx,'removed '+bet_key+' by '+nick)
+                    self.lock = False
+                    return
+            except KeyError as e:
```

Comment:
hmMm this seems like a hack. can you add relevant key checks and omit the try-catch?